Scripts/Game/UI/Modded/COA_CompassUI.c

```diff
@@ -56,241 +51,64 @@ class COA_Compass : SCR_InfoDisplay
-			if (!statusDisplay || !playerDisplay || !playerName || !localplayer) continue;
-			
-			// Set player text.
-			playerDisplay.SetText(playerName);
-			
-			//---------------------------------------------------------------------------------------------------------------------------------
-			// Color Teams
-			//---------------------------------------------------------------------------------------------------------------------------------
-			
-			//statusDisplay.SetColorInt(ARGB(255, 255, 160, 160))
-			
-			//---------------------------------------------------------------------------------------------------------------------------------
-			// Vehicle Icons, they supercede any specialty icons
-			//---------------------------------------------------------------------------------------------------------------------------------
-			
-			// Check if target player is even in a vehicle.
-			CompartmentAccessComponent compartmentAccess = CompartmentAccessComponent.Cast(localplayer.FindComponent(CompartmentAccessComponent));
-			if (compartmentAccess)
-			{
-				// Check target players current compartment.
-			   BaseCompartmentSlot compartment = compartmentAccess.GetCompartment();  
-			    if (compartment)
-			    {
-					// Check target players current compartment type, then assign his icon.
-			        ECompartmentType compartmentType = SCR_CompartmentAccessComponent.GetCompartmentType(compartment);
-					switch (compartmentType)
-					{
-						// Passanger/Commander 
-						// ToDo: impliment Commander compartment detection and icon when/if it's added to the game
-						case ECompartmentType.Cargo : {
-							statusDisplay.SetOpacity(1);
-							statusDisplay.LoadImageTexture(0, "{B910A93F355F168C}Layouts\UI\Textures\Icons\imagecargo_ca.edds");
-							break;
-						}
-						// Driver
-						case ECompartmentType.Pilot : {
-							statusDisplay.SetOpacity(1);
-							statusDisplay.LoadImageTexture(0, "{C2B2F451FB157A89}Layouts\UI\Textures\Icons\imagedriver_ca.edds");
-							break;
-						}
-						// Gunner
-						case ECompartmentType.Turret : {
-							statusDisplay.SetOpacity(1);
-							statusDisplay.LoadImageTexture(0, "{3DAAB773C8C29812}Layouts\UI\Textures\Icons\imagegunner_ca.edds");
-							break;
-						}
-					}
-					// Skip ahead to next for-loop iteration since we've done our job of setting the target players icon.
-					continue;
-			    };
-			};
+			int playerPlace = 0;
 
-			//---------------------------------------------------------------------------------------------------------------------------------
-			//	Specialty Icons
-			//---------------------------------------------------------------------------------------------------------------------------------
-			
-			// Check if current target player is the current squad leader AND we're at the correct position.
-			if (localPlayerID == groupLeaderID && i == 0) 
+			// Parse through current group array.
+			for(int i = 0; i<groupCount && i < 12; i++) 
```

Comment:
Can we constant the max group size at the top of this file? Like:

`const int MAXGROUPSIZE = 12;`

Then in the for loop on line 73 here:

`for(int i = 0; i<MAXGROUPSIZE; i++) `

Reply:
Yeah, we can, but since there's no support for any other group size, it's null rn

also, for loop would look like this, just so you don't get confused:
```for(int i = 0; i<groupCount && i < MAXGROUPSIZE; i++)```